objects/src/accounts/storage/mod.rs

```diff
@@ -267,193 +82,80 @@ impl AccountStorage {
     ///
     /// This method assumes that the delta has been validated by the calling method and so, no
     /// additional validation of delta is performed.
-    ///
-    /// Returns an error if:
-    /// - The delta implies an update to a reserved account slot.
-    /// - The updates violate storage layout constraints.
-    /// - The updated value has an arity different from 0.
-    pub(super) fn apply_delta(&mut self, delta: &AccountStorageDelta) -> Result<(), AccountError> {
-        // --- update storage maps --------------------------------------------
-
-        for &(slot_idx, ref map_delta) in delta.updated_maps.iter() {
-            let storage_map =
-                self.maps.get_mut(&slot_idx).ok_or(AccountError::StorageMapNotFound(slot_idx))?;
-
-            let new_root = storage_map.apply_delta(map_delta)?;
-
-            let index = LeafIndex::new(slot_idx.into()).expect("index is u8 - index within range");
-            self.slots.insert(index, new_root.into());
+    pub(super) fn apply_delta(&mut self, delta: &AccountStorageDelta) {
+        for (idx, storage_slot) in delta.items() {
+            self.set_item(*idx, storage_slot.clone())
         }
-
-        // --- update storage slots -------------------------------------------
-
-        for &slot_idx in delta.cleared_items.iter() {
-            self.set_item(slot_idx, Word::default())?;
-        }
-
-        for &(slot_idx, slot_value) in delta.updated_items.iter() {
-            self.set_item(slot_idx, slot_value)?;
-        }
-
-        Ok(())
     }
 
     /// Updates the value of the storage slot at the specified index.
-    ///
-    /// This method should be used only to update simple value slots. For updating values
-    /// in storage maps, please see [AccountStorage::set_map_item()].
-    ///
-    /// # Errors
-    /// Returns an error if:
-    /// - The index specifies a reserved storage slot.
-    /// - The update tries to set a slot of type array.
-    /// - The update has a value arity different from 0.
-    pub fn set_item(&mut self, index: u8, value: Word) -> Result<Word, AccountError> {
-        // layout commitment slot cannot be updated
-        if index == Self::SLOT_LAYOUT_COMMITMENT_INDEX {
-            return Err(AccountError::StorageSlotIsReserved(index));
-        }
+    pub fn set_item(&mut self, index: u8, storage_slot: StorageSlot) {
```

Comment:
We should keep the signature of this method as it was before. Basically, we want to keep mutators parallel to their counterparts in the MASM code.